File: netpulse-api/internal/incidents/resolution_test.go
package incidents

import "testing"

func TestCannotResolveFromOneRecovery(t *testing.T) {
	got := CanMarkResolved(ResolutionInput{RecoverySampleCount: 1, IdentifiedCause: true})
	if got.OK {
		t.Fatal("a single recovered measurement must not resolve an incident")
	}
}

func TestCannotResolveWithoutIdentifiedCause(t *testing.T) {
	got := CanMarkResolved(ResolutionInput{RecoverySampleCount: 4, IdentifiedCause: false})
	if got.OK {
		t.Fatal("recoveries without an identified cause must not resolve")
	}
}

func TestResolveNeedsMultipleRecoveriesAndCause(t *testing.T) {
	got := CanMarkResolved(ResolutionInput{RecoverySampleCount: 2, IdentifiedCause: true})
	if !got.OK {
		t.Fatal(got.Reason)
	}
}

File: netpulse-web/src/features/observatory/resolution.test.ts
import { describe, expect, it } from "vitest";

import { canMarkResolved } from "@/features/observatory/resolution";

describe("incident resolution policy", () => {
  it("never marks resolved from one recovered measurement", () => {
    const decision = canMarkResolved({
      recoverySampleCount: 1,
      identifiedCause: true,
    });
    expect(decision.ok).toBe(false);
  });

  it("requires an identified cause in addition to recoveries", () => {
    expect(
      canMarkResolved({ recoverySampleCount: 4, identifiedCause: false }).ok
    ).toBe(false);
  });
});
